src/app/Shared/Services/CommandChannel.service.tsx

```diff
@@ -35,17 +35,18 @@ export class CommandChannel {
 
     fromFetch(`${this.apiSvc.authority}/health`)
       .pipe(
-        concatMap(resp => from(resp.json())))
-      .pipe(
-        concatMap(jsonResp => {
-            if((jsonResp.dashboardAvailable == true) && (jsonResp.datasourceAvailable == true)) 
-              {return forkJoin([getDatasourceURL, getDashboardURL])}
-            else {this.logError('Grafana configuration', 'URLs unavailable');
-                  return jsonResp}}))
+        concatMap(resp => from(resp.json())), 
```

Comment:
Did the `concatMap(resp => from(resp.json()))` -> `map(resp => resp.json())` transform not apply here?

Reply:
So Ive been trying to figure out why it doesn't work there or above in the getDatasourceURL either, and I think it is because the concatMap waits for the resp.json() to finish before proceding, which the map doesn't. Im not 100% sure this is why, but when I tried changing it it would not parse the URLs

Reply:
Strange. I'm not sure if it's because of the `.json()` "not finishing", but maybe something about how the `fromFetch` emits values and when it emits (or doesn't) its completion event. It's fine to leave it with the `concatMap` in that case, if that's the way that works.